### DISCUSSION TITLE
get template from git

### DISCUSSION THREAD

**Discussion by trondhindenes** [PowerShellOrg/Plaster#381] [Ideas]
It would be great if Plaster natively supported git-based templates (something like git:https://github.com/trondhindenes/plastertemplate1.git so that the template content was disconnected from the module itself - would require some convention around how to store stuff in a git repo in a way that Plaster would understand (esp content files etc)

**Comment by MartinSGill** [comment]
I've pretty much written a Plaster module in this manner already.

As long as the conventions of a plaster template are followed as already documented, it's very easy to treat a repository as a template. It even works right now, with a couple of manual steps.

```powershell
git clone --depth 1 https://github.com/SomeUser/SomeTemplate
Invoke-Plaster -TemplatePath .\SomeTemplate -DestinationPath PathToNewModule
```

To implement this in Plaster all you'd really need, if I quickly think about it, is

- A new Parameter (e.g. -GitHub [URL | User/Repo] )
- A place to cache the clone, e.g. ProgramData/Plaster

Extra credit for:

- Keeping the clone up-to-date
- Error handling around git not being available
